airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -158,6 +163,9 @@ def _resolve_connection(self):
             conn_data['deploy_mode'] = extra.get('deploy-mode', None)
             conn_data['spark_home'] = extra.get('spark-home', None)
             conn_data['spark_binary'] = extra.get('spark-binary', 'spark-submit')
+            # Only define namespace property if it has been provided
+            if extra.get('namespace'):
+                conn_data['namespace'] = extra.get('namespace')
```

Comment:
Suggestion: `conn_data['namespace'] = extra.get('namespace', 'default')`

Reply:
I had that originally, but given it only enters this  if namespace is defined it would be pointless.  I could define namespace to None/default for all schedule types (big impact on tests), or I could move the test for K8s master to here and only do this section of K8s mode? I should also set the spark.kubernetes.namespace --conf property if namespace is defined I think.

Reply:
In the initial it would always run in the default namespace:
```
+                    api_response = client.delete_namespaced_pod(
+                        self._kubernetes_driver_pod,
+                        "default",
```
What I read from the docs, it isn't optional, so I'm not sure if `None` would work.:
https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#delete_namespaced_pod

The suggestion that I had was to do this in a single line, with a default value, instead of the if-else construction.

I strongly believe starting (and deleting) a job using a configurable namespace should be in there. If you are using a single kubernetes cluster for one person/team, only using the default namespace would be ok. But having namespaces in the connection strings, allows us to kubernetes cluster among different teams/projects very nicely with the configured resources quota's.

We need to supply this when supply this to Spark when starting the job: https://spark.apache.org/docs/latest/running-on-kubernetes.html#namespaces
And supply the same namespace when deleting the pod, otherwise Kubernetes cannot find the resource to delete.

Do you agree? :)

Reply:
On second thoughts, will just modify the mesos, yarn tests

Reply:
AFAIK, dicts are always unordered, the tests should compare the keys. If this requires changes, please do.